qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -114,6 +114,26 @@ def get_monitor_layout():
     return outputs
 
 
+def set_keyboard_layout(vm):
+    """Set layout configuration into features for Gui admin extension"""
+    try:
+        # Example of 'xprop -root _XKB_RULES_NAMES' output:
+        # _XKB_RULES_NAMES(STRING) = "evdev", "pc105", "fr", "oss", ""
+        xkb_rules_names = subprocess.check_output(
+            ['xprop', '-root', '_XKB_RULES_NAMES']).decode()
+        xkb_rules_names = xkb_rules_names. \
+            replace('"', '').replace('\n', '')
+        keyboard_layout = '+'.join(xkb_rules_names.split(', ')[2:])
```

Comment:
I think this will break if multiple layouts are set. Example:
```
$ xprop -root _XKB_RULES_NAMES
_XKB_RULES_NAMES(STRING) = "evdev", "pc105", "pl,us", ",", "grp:win_switch,compose:caps"
```

Note layout and variant behave differently from options: the former ones are matching lists, but options are global. Note the coma in variant value. The above value translated is:
 - "pl" layout with empty variant and "grp:win_switch,compose:caps" options
 - "us" layout with empty variant and "grp:win_switch,compose:caps" options

What is expected feature value here?

Reply:
In fact, since there is no support for dynamic layout notification yet, it may be better to extract info just about the current layout. Information about secondary layouts is useless without info when it is switched. And when we get dynamic change, we can simply replace the whole value with the changed layout - still, there is no need to propagate secondary (inactive) layouts outside of GUI VM itself (where it is needed for layout changing widget).

Possibly tricky question is which of those is the current one - I have no idea how to get this information. For now I'd approximate with the first one.

Reply:
This is going to break as I was not aware of such kind of possible values. I'll use match with regex and extract only first values (first layout).